src/buffer.c

```diff
@@ -38,6 +38,55 @@ BUFFER_HANDLE BUFFER_new(void)
     return (BUFFER_HANDLE)temp;
 }
 
+// return 0 if safemalloc succeeds else return non-zero
+int BUFFER_safemalloc(BUFFER_HANDLE* handleptr, size_t size)
+{
+    int result;
+    // if handleptr or handle is NULL, return immediately
+    if (!handleptr || !(*handleptr))
+    {
+        result = __LINE__;
+    }
+    else
+    {
+        if (size == 0)
+        {
+            (*handleptr)->buffer = (unsigned char*)malloc(1);
+            if ((*handleptr)->buffer == NULL)
+            {
+                /*Codes_SRS_BUFFER_02_003: [If allocating memory fails, then BUFFER_create shall return NULL.]*/
+                free(*handleptr);
+                *handleptr = NULL;
+                result = __LINE__;
+            }
+            else
+            {
+                (*handleptr)->buffer[0] = '\0';
```

Comment:
>buffer[0] = '\0'; [](start = 30, length = 17)

No need to initialize the memory to anything. This will allow you to have one code path for calling malloc.